src/transport/Session.h

```diff
@@ -225,6 +228,11 @@ class Session
 
     bool IsUnauthenticatedSession() const { return GetSessionType() == SessionType::kUnauthenticated; }
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    ActiveConnectionState * GetTCPConnection() const { return mTCPConnection; }
+    void SetTCPConnection(ActiveConnectionState * conn) { mTCPConnection = conn; }
```

Comment:
Having this Set() be callable any time is a bit odd, but I guess sessions can migrate to new connections?

What is the lifetime situation here?  How do we avoid dangling pointers to ActiveConnectionState?

Reply:
The lifetime assumptions need to be clearly documented.